packages/web/src/components/track/GiantTrackTile.tsx

```diff
@@ -0,0 +1,655 @@
+              <div className={cn(fadeIn)}>
+                <span>By</span>
+                <ArtistPopover handle={artistHandle}>
+                  <h2 className={styles.artist} onClick={onClickArtistName}>
+                    {artistName}
+                    <UserBadges
+                      className={styles.verified}
+                      badgeSize={18}
+                      userId={userId}
+                    />
+                  </h2>
+                </ArtistPopover>
+              </div>
+              {isLoading && (
+                <Skeleton className={styles.skeleton} width='60%' />
+              )}
+            </div>
+          </div>
+
+          <div className={cn(styles.playSection, fadeIn)}>
+            <PlayPauseButton
+              doesUserHaveAccess={doesUserHaveAccess}
+              playing={playing}
+              onPlay={onPlay}
+              trackId={trackId}
+            />
+            {isLongFormContent && isNewPodcastControlsEnabled ? (
+              <GiantTrackTileProgressInfo
+                duration={duration}
+                trackId={trackId}
+              />
+            ) : (
+              renderListenCount()
+            )}
+          </div>
+
+          <div className={cn(styles.statsSection, fadeIn)}>
+            {renderStatsRow()}
+          </div>
+
+          <div
+            className={cn(styles.actionButtons, fadeIn)}
+            role='group'
+            aria-label={messages.actionGroupLabel}
+          >
+            {renderShareButton()}
+            {renderMakePublicButton()}
+            {doesUserHaveAccess && renderRepostButton()}
+            {doesUserHaveAccess && renderFavoriteButton()}
+            <span>
+              {/* prop types for overflow menu don't work correctly
+              so we need to cast here */}
+              <Menu {...(overflowMenu as any)}>
+                {(ref, triggerPopup) => (
+                  <div className={cn(styles.menuKebabContainer)} ref={ref}>
+                    <Button
+                      className={cn(styles.buttonFormatting, styles.moreButton)}
+                      leftIcon={<IconKebabHorizontal />}
+                      onClick={triggerPopup}
+                      text={null}
+                      textClassName={styles.buttonTextFormatting}
+                      type={ButtonType.COMMON}
+                    />
+                  </div>
+                )}
+              </Menu>
+            </span>
+          </div>
+        </div>
+        <div className={styles.badges}>
+          {aiAttributionUserId ? (
+            <Badge
+              icon={<IconRobot />}
+              className={styles.badgeAi}
+              textLabel={messages.generatedWithAi}
+            />
+          ) : null}
+          {badge ? (
+            <Badge className={styles.badgePlacement} textLabel={badge} />
+          ) : null}
+        </div>
+      </div>
+
+      {isPremium && premiumConditions ? (
```

Comment:
Added a check here on premium conditions to avoid rendering this section altogether. Technically I think the child component is safe if it isn't passed. But in `TrackTile` we follow the pattern of not rendering the section if there are no premium conditions.